src/decoder/wav.rs

```diff
@@ -65,32 +69,38 @@ impl<R> Iterator for SamplesIterator<R>
 where
     R: Read + Seek,
 {
-    type Item = i16;
+    type Item = DecoderFormat;
 
     #[inline]
-    fn next(&mut self) -> Option<i16> {
+    fn next(&mut self) -> Option<Self::Item> {
+        self.samples_read += 1;
         let spec = self.reader.spec();
         match (spec.sample_format, spec.bits_per_sample) {
-            (SampleFormat::Float, 32) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                f32_to_i16(value.unwrap_or(0.0))
-            }),
-            (SampleFormat::Int, 8) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i8_to_i16(value.unwrap_or(0))
-            }),
-            (SampleFormat::Int, 16) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                value.unwrap_or(0)
-            }),
-            (SampleFormat::Int, 24) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i24_to_i16(value.unwrap_or(0))
-            }),
-            (SampleFormat::Int, 32) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i32_to_i16(value.unwrap_or(0))
-            }),
+            (SampleFormat::Float, 32) => self
+                .reader
+                .samples()
+                .next()
+                .map(|value| value.unwrap_or(0.0)),
```

Comment:
// I think we can use dasp_sample::Sample::EQUILIBRIUM as default.

Reply:
Will do. Note that there are more places in Rodio that could use this, in particular all the implementations of [`Sample::zero_value`](https://github.com/RustAudio/rodio/blob/ea45c17db104f27645972e0f68bfa0836cf9d04e/src/conversions/sample.rs#L93). I can add that with this PR.

Reply:
Mostly done in [a1cf45a](https://github.com/RustAudio/rodio/pull/669/commits/a1cf45a02c723043523a017a55386779023e2c70). I am sure there is more to clean up, like `Sample::saturating_add`, but I need to get to bed :wink: so I'll get to that later.